src/solvers/flattening/boolbv_abs.cpp

```diff
@@ -16,24 +16,20 @@ Author: Daniel Kroening, kroening@kroening.com
 
 bvt boolbvt::convert_abs(const exprt &expr)
 {
-  std::size_t width=boolbv_width(expr.type());
+  const std::size_t width = boolbv_width(expr.type());
 
   if(width==0)
     return conversion_failed(expr);
 
   const exprt::operandst &operands=expr.operands();
-
-  if(operands.size()!=1)
-    throw "abs takes one operand";
-
+  DATA_INVARIANT(operands.size() == 1, "abs takes one operand");
```

Comment:
May I suggest that instead the interface be changed to take a `const abs_exprt &`, which will ensure this property.